source/Applications/Advanced/HandEyeCalibration/HandEyeCalibration/HandEyeCalibration.cpp

```diff
@@ -70,6 +70,41 @@ namespace
         const auto settings = Zivid::CaptureAssistant::suggestSettings(camera, parameters);
         return camera.capture(settings);
     }
+
+    std::string enterTypeOfCalibration()
+    {
+        while(true)
+        {
+            std::cout << "Enter command, eth (for eye-to-hand) or eih (eye-in-hand): ";
+            const auto calibration = getInput();
+
+            if(calibration == "eth" || calibration == "ETH")
+            {
+                return "eth";
+            }
+            if(calibration == "eih" || calibration == "EIH")
+            {
+                return "eih";
+            }
+            std::cout << "Entered uknown method" << std::endl;
+        }
+    }
+
+    Zivid::Calibration::HandEyeOutput performCalibration(const std::vector<Zivid::Calibration::HandEyeInput> &input,
+                                                         const std::string &calibrationType)
+    {
+        if(calibrationType == "eth")
+        {
+            std::cout << "Performing eye-to-hand calibration" << std::endl;
+            return Zivid::Calibration::calibrateEyeToHand(input);
+        }
+        if(calibrationType == "eih")
+        {
+            std::cout << "Performing eye-in-hand calibration" << std::endl;
+            return Zivid::Calibration::calibrateEyeInHand(input);
+        }
+        throw std::invalid_argument("Unknown calibration method");
```

Comment:
Shouldn't this line of code be after line 89 - in the same method?

Reply:
you mean merge the two functions: enterTypeOfCalibration() and performCalibration() ?
So that it asks for clibration method when performing it?
If not I'm not sure what you mean